src/compiler/scala/tools/nsc/typechecker/Unapplies.scala

```diff
@@ -42,10 +43,31 @@ trait Unapplies extends ast.TreeDSL {
    */
   def directUnapplyMember(tp: Type): Symbol = (tp member nme.unapply) orElse (tp member nme.unapplySeq)
 
-  /** Filters out unapplies with multiple (non-implicit) parameter lists,
-   *  as they cannot be used as extractors
+  /** Filters out unapplies with invalid shapes: extractor methods must have
+    * either one unary param list or one unary param list and an implicit param list.
    */
-  def unapplyMember(tp: Type): Symbol = directUnapplyMember(tp) filter (sym => !hasMultipleNonImplicitParamLists(sym))
```

Comment:
it seems `hasMultipleNonImplicitParamLists` is unused at this piont, let's remove it.